File: CX/ClientApp/src/servicios/ServicioEtapasTallerCo.js
import { ProcesarDatosApi } from "./ApiFetch";
const controlador = "etapasTallerCo";

export const AgregarEtapaTallerCo = async (data) => {
    const url = `${controlador}/AgregarEtapaTallerCo`;
    return await ProcesarDatosApi('POST', url, data);
}

export const ActualizarEtapaTallerCo = async (data) => {
    const url = `${controlador}/ActualizarEtapaTallerCo`;
    return await ProcesarDatosApi('PUT', url, data);
}

export const ObtenerEtapaTallerCoPorID = async (id) => {
    const url = `${controlador}/ObtenerEtapaTallerCoPorID/${id}`;
    return await ProcesarDatosApi('GET', url);
}

export const ObtenerEtapasTallerCo = async (id, id1) => {
    const url = `${controlador}/ObtenerEtapasTallerCo/${id}/${id1}`;
    return await ProcesarDatosApi('GET', url);
}
export const ObtenerFechasTallerCo = async (id, id1) => {
    const url = `${controlador}/ObtenerFechasTallerCo/${id}/${id1}`;
    return await ProcesarDatosApi('GET', url);
}
export const ObtenerCantidadEtapasTallerCo = async () => {
    const url = `${controlador}/ObtenerCantidadEtapasTallerCo`;
    return await ProcesarDatosApi('GET', url);
}

File: BussinesLogic/Servicios/ServicioEtapasTallerCo.cs
using DataAccess.StoredProcedures;
using Dominio.Dto;
using Dominio.Entiti;
using Dominio.Interfaces.Aplicacion.EtapaTallerCo;
using Dominio.Interfaces.Infraestructura.BaseDatos;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace BussinesLogic.Servicios
{
    public class ServicioEtapasTallerCo : IServicioEtapaTallerCo
    {

        private readonly IRepositorioEtapaTallerCo repositorioEtapaTallerCo;
        public ServicioEtapasTallerCo(IRepositorioEtapaTallerCo repositorioEtapaTallerCo)
        {
            this.repositorioEtapaTallerCo = repositorioEtapaTallerCo;
        }

        public async Task<DtoRespuestaSP> AgregarEtapaTallerCo(EntitiEtapaTallerCo entitiEtapaTallerCo)
        {
            return await this.repositorioEtapaTallerCo.AgregarEtapaTallerCo(entitiEtapaTallerCo);
        }
        public async Task<DtoRespuestaSP> ActualizarEtapaTallerCo(EntitiEtapaTallerCo entitiEtapaTallerCo)
        {
            return await this.repositorioEtapaTallerCo.ActualizarEtapaTallerCo(entitiEtapaTallerCo);
        }

        public async Task<DtoEtapaTallerCo> ObtenerEtapaTallerCoPorID(int idEtapaTallerCo)
        {
            return await this.repositorioEtapaTallerCo.ObtenerEtapaTallerCoPorID(idEtapaTallerCo);
        }

        public async Task<List<DtoEtapaTallerCo>> ObtenerEtapasTallerCo(int idTipoTaller, int idTallerCoCreacion)
        {
            return await this.repositorioEtapaTallerCo.ObtenerEtapasTallerCo(idTipoTaller, idTallerCoCreacion);
        }
        public async Task<DtoEtapaTallerCo> ObtenerFechasTallerCo(int idTipoTaller, int idTallerCoCreacion)
        {
            return await this.repositorioEtapaTallerCo.ObtenerFechasTallerCo(idTipoTaller, idTallerCoCreacion);
        }

        public async Task<DtoCantidadDatos> ObtenerCantidadEtapasTallerCo()
        {
            return await this.repositorioEtapaTallerCo.ObtenerCantidadEtapasTallerCo();
        }
    }
}